File: lib/src/mock/mock_application_ref.dart
library angular2.src.mock.mock_application_ref;

import "package:angular2/src/core/application_ref.dart" show ApplicationRef;
import "package:angular2/src/core/di.dart" show Injectable;
import "package:angular2/src/facade/lang.dart" show Type;
import "package:angular2/src/core/linker/dynamic_component_loader.dart"
    show ComponentRef;
import "package:angular2/src/core/di.dart" show Provider, Injector;
import "package:angular2/src/core/zone/ng_zone.dart" show NgZone;
import "package:angular2/src/facade/async.dart" show Future;

/**
 * A no-op implementation of [ApplicationRef], useful for testing.
 */
@Injectable()
class MockApplicationRef extends ApplicationRef {
  void registerBootstrapListener(
      dynamic /* (ref: ComponentRef) => void */ listener) {}
  void registerDisposeListener(dynamic /* () => void */ dispose) {}
  Future<ComponentRef> bootstrap(Type componentType,
      [List<dynamic /* Type | Provider | List < dynamic > */ > bindings]) {
    return null;
  }

  Injector get injector {
    return null;
  }

  NgZone get zone {
    return null;
  }

  void dispose() {}
  void tick() {}
  List<Type> get componentTypes {
    return null;
  }
}

File: npm/angular2/ts/src/mock/mock_application_ref.ts
import {ApplicationRef} from 'angular2/src/core/application_ref';
import {Injectable, Injector} from 'angular2/src/core/di';
import {Type} from 'angular2/src/facade/lang';
import {ComponentRef, ComponentFactory} from 'angular2/src/core/linker/component_factory';
import {NgZone} from 'angular2/src/core/zone/ng_zone';

/**
 * A no-op implementation of {@link ApplicationRef}, useful for testing.
 */
@Injectable()
export class MockApplicationRef extends ApplicationRef {
  registerBootstrapListener(listener: (ref: ComponentRef<any>) => void): void {}

  registerDisposeListener(dispose: () => void): void {}

  bootstrap<C>(componentFactory: ComponentFactory<C>): ComponentRef<C> { return null; }

  get injector(): Injector { return null; };

  get zone(): NgZone { return null; };

  run(callback: Function): any { return null; }

  waitForAsyncInitializers(): Promise<any> { return null; }

  dispose(): void {}

  tick(): void {}

  get componentTypes(): Type[] { return null; };
}
